### ISSUE TITLE
jupyter notebook shoudl open current server if existing.

### ISSUE THREAD

**Issue by Carreau** [ipython/ipython#9342]
Came up during Jupyter Days education panel, 

Apparently many student do not realise that `Jupyter notebook` start a new server, and once they close their last browser tab, they re-issue the command in a new terminal, leading to tens of started servers. 

Some people thus ask wether `jupyter notebook` without parameters could be switched to open the browser if a current server is already opened. This would mean potentially making starting a (new) server a separate flag. 

ping @ellisonbg

**Comment by Carreau** [comment]
I'm stupid, I don't know why I opened that here, like #9341 I'm tired .I'll migrate that later.

**Comment by takluyver** [comment]
I already have the code to do that in nbopen - I'm happy for it to get merged into the notebook itself.

**Comment by Carreau** [comment]
Yeah, I think this is more a question of how to keep backward compatibility. 
We likely want:

5.0 - current behavior with warning. 
6.0 - new behavior. 

ideas: 
- `jupyter notebook-server` to actually start a server ? 
- `jupyter notebook start|stop|restart` ?

**Comment by takluyver** [comment]
I wouldn't worry too much about backwards compatibility. The only reason to run `jupyter notebook foo.ipynb` is to open that notebook in a browser, so it's probably not written into scripts much. I'd just make that check for an existing server and use it if available. We could have a `--new-server` flag or something to force it to start a new server. This is already how many programs (e.g. browsers) work.

Having some kind of management interface probably makes sense (see also nbmanager), but it's more than just start/stop/restart, because there can be multiple servers running.
